### DISCUSSION TITLE
How could Crescendo utilize JC (JSON Convert) or it's core concept?

### DISCUSSION THREAD

**Discussion by aaronsb** [PowerShell/Crescendo#157] [Ideas]
Perhaps my use case is too much of a corner case, but I thought I'd bring it up here.

I typically use PSCore on my linux desktop for my shell, (I know, weird right?) and therefore I have several command wrappers for things like mount, lsusb, systemctl, and others. In order to get valuable object oriented output, I pipe these through [jc](https://github.com/kellyjonbrazil/jc), which json parses the output from a lot of these tools.

Recently I've started using Crescendo to map inputs back to these tools to build a common set of unix/linux commands (because I like objects instead of strings? or perhaps I just like doing it as an exercise) building a hybrid set of functions.

My question or core discussion topic is this: does it make sense to recognize an intermediary like JC as part of the process of "compiling" various Cresecndo-ized commands? This greatly simplifies the parsing and prerequisites to build a Crescendo package, but introduces another dependency.

Or, thinking about it another way, does it make sense to begin to maintain a standardized or recognized library of string parsers for known common commands, so it doesn't feel like we're always "starting from scratch" with parsing complex string output commands?

**Comment by theJasonHelmick** [comment]
I think the idea of using jc is interesting.  Could you provide an example?
